### PR TITLE
Added tests for updating an Activation Key's auto-attach value

### PR DESCRIPTION
**Pull request by JacobCallahan**
nosetests -v tests/foreman/cli/test_activationkey.py
@Test: Attempt to update Activation key with bad auto-attach value ... ok
@Test: Update Activation key with inverse auto-attach value ... ok
@Test: Update Activation key with valid auto-attach values ... ok
@Test: Update Activation key with valid auto-attach values ... ok
@Test: Update Activation key with valid auto-attach values ... ok
@Test: Update Activation key with valid auto-attach values ... ok
@Test: Update Activation key with valid auto-attach values ... ok
@Test: Update Activation key with valid auto-attach values ... ok

Ran 8 tests in 72.560s

OK

### CODE REVIEW

File: robottelo/cli/activationkey.py [hunk 1]
```diff
@@ -47,14 +49,28 @@ def add_subscription(cls, options=None):
...
+    def content_override(cls, options=None):
+        """Override product content defaults"""
+        cls.command_sub = 'content-override'
+        return cls.execute(cls._construct_command(options))
+
+    @classmethod
     def copy(cls, options=None):
         """Copy an activation key"""
         cls.command_sub = 'copy'
+        return cls.execute(cls._construct_command(options))
```
Comment (Ichimonji10) [inline robottelo/cli/activationkey.py]: Nice catch.

Comment (Ichimonji10) [inline robottelo/cli/activationkey.py]: Also, good catch on line 67.

File: tests/foreman/cli/test_activationkey.py [hunk 2]
```diff
@@ -1575,3 +1575,106 @@ def test_positive_copy_subscription(self):
...
             result.stdout[3]  # subscription list
         )
+
+    def test_positive_update_auto_attach(self):
+        """@Test: Update Activation key with inverse auto-attach value
+
+        @Feature: Activation key update / info
+
+        @Steps:
+        1. Get the key's current auto attach value.
```
Comment (Ichimonji10) [inline tests/foreman/cli/test_activationkey.py]: Add a blank line between `@steps` and the list.

File: tests/foreman/cli/test_activationkey.py [hunk 3]
```diff
@@ -1575,3 +1575,106 @@ def test_positive_copy_subscription(self):
...
+        u'false',
+        u'yes',
+        u'no',
+    )
+    def test_positive_update_auto_attach_2(self, new_value):
+        """@Test: Update Activation key with valid auto-attach values
+
+        @Feature: Activation key update / info
+
+        @Steps:
```
Comment (Ichimonji10) [inline tests/foreman/cli/test_activationkey.py]: Add a blank line between `@steps` and the list. Without that line, this is considered one big paragraph:

    @Steps: 1. Update the key with a valid value 2. Verify key was updated

File: tests/foreman/cli/test_activationkey.py [hunk 4]
```diff
@@ -1575,3 +1575,106 @@ def test_positive_copy_subscription(self):
...
+        self.assertEqual(result.return_code, 0)
+        self.assertEqual(
+            u'Activation key updated', result.stdout[0]['message'])
+
+    def test_negative_update_auto_attach(self):
+        """@Test: Attempt to update Activation key with bad auto-attach value
+
+        @Feature: Activation key update / info
+
+        @Steps:
```
Comment (Ichimonji10) [inline tests/foreman/cli/test_activationkey.py]: Same.

**Comment by elyezer** [comment]
ACK, pending @Ichimonji10's comments.

File: tests/foreman/cli/test_activationkey.py [hunk 5]
```diff
@@ -1575,3 +1575,106 @@ def test_positive_copy_subscription(self):
...
+
+        """
+        org_id = make_org(cached=True)['id']
+        key_id = make_activation_key(
+            {u'organization-id': org_id}, cached=True)['id']
+        try:
+            result = ActivationKey.update({
+                u'id': key_id,
+                u'organization-id': org_id,
+                u'auto-attach': gen_string('utf8')
```
Comment (Ichimonji10) [inline tests/foreman/cli/test_activationkey.py]: Nitpick: Add a trailing comma and and alphabetically sort these.

File: tests/foreman/cli/test_activationkey.py [hunk 6]
```diff
@@ -1575,3 +1575,106 @@ def test_positive_copy_subscription(self):
...
+
+        """
+        org_id = make_org(cached=True)['id']
+        key_id = make_activation_key(
+            {u'organization-id': org_id}, cached=True)['id']
+        try:
+            result = ActivationKey.update({
+                u'id': key_id,
+                u'organization-id': org_id,
+                u'auto-attach': new_value
```
Comment (Ichimonji10) [inline tests/foreman/cli/test_activationkey.py]: Nitpick: Add a trailing comma and and alphabetically sort these.

File: tests/foreman/cli/test_activationkey.py [hunk 7]
```diff
@@ -1575,3 +1575,106 @@ def test_positive_copy_subscription(self):
...
+            }).stdout['auto-attach']
+        except CLIFactoryError as err:
+            self.fail(err)
+        # invert value
+        new_value = u'false' if attach_value == u'true' else u'true'
+        try:
+            result = ActivationKey.update({
+                u'id': key_id,
+                u'organization-id': org_id,
+                u'auto-attach': new_value
```
Comment (Ichimonji10) [inline tests/foreman/cli/test_activationkey.py]: Nitpick: Add a trailing comma and and alphabetically sort these.

**Comment by Ichimonji10** [comment]
Looks good. ACK pending comments.

File: tests/foreman/cli/test_activationkey.py (hunk 5 above)
Comment (elyezer) [inline tests/foreman/cli/test_activationkey.py]: :+1:

**Comment by JacobCallahan** [comment]
comments addressed and all tests pass. Thanks for the input!

**Pull request merged**